tracer/src/Datadog.Trace.SourceGenerators/EnumExtensions/Sources.cs

```diff
@@ -0,0 +1,213 @@
+﻿// <copyright file="Sources.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Text;
+
+namespace Datadog.Trace.SourceGenerators.EnumExtensions;
+
+internal class Sources
+{
+    public const string Attributes = """
+        // <auto-generated />
+        #nullable enable
+
+        namespace Datadog.Trace.SourceGenerators;
+
+        /// <summary>
+        /// Add to enums to indicate that extension methods should be generated for the type
+        /// </summary>
+        [System.AttributeUsage(System.AttributeTargets.Enum)]
+        [System.Diagnostics.Conditional("DEBUG")]
+        internal class EnumExtensionsAttribute : System.Attribute
+        {
+        }
```

Comment:
Why not😂 

It doesn't _need_ to be, it's just a common convention with source generators like this, as it works well if (for example) you wanted to put it into a NuGet package. By using the `RegisterPostInitializationOutput()` hook, you can ensure the attribute is always available (instead of the user having to add their own). 

It makes things like unit/integration tests easier too, as otherwise you have to manually add this attribute definition everywhere, make sure to get the namespace correct, etc.

However, there are arguments for defining it directly ([see my "marker attributes problem" posts](https://andrewlock.net/creating-a-source-generator-part-7-solving-the-source-generator-marker-attribute-problem-part1/))! I'm inclined to leave it as-is for now, if it's a problem later we can easily define it directly instead.

Reply:
Sounds good to me, wasn't sure if there was some specific reason for it.

Also, you've got blog posts for everything it seems 😆